backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Conditions_ModularContract_Clone.java

```diff
@@ -44,23 +41,6 @@ protected String doIt()
 	{
 		final I_C_Year newYear = InterfaceWrapperHelper.load(YearId.ofRepoId(p_C_Year_ID), I_C_Year.class);
 		final I_C_Flatrate_Conditions conditions = flatrateDAO.getConditionsById(ConditionsId.ofRepoId(getRecord_ID()));
-		final I_ModCntr_Settings settings = conditions.getModCntr_Settings();
-
-		Check.assumeNotNull(settings, " Should never happen because settings are mandatory for modular contract conditions", conditions);
-
-		final ProductId productId = ProductId.ofRepoId(settings.getM_Product_ID());
-		final YearAndCalendarId yearAndCalendarId = YearAndCalendarId.ofRepoIdOrNull(p_C_Year_ID, settings.getC_Calendar_ID());
-
-		final ModularContractSettingsQuery query = ModularContractSettingsQuery.builder()
-				.productId(productId)
-				.yearAndCalendarId(yearAndCalendarId)
-				.build();
-
-		if (modularContractSettingsDAO.isSettingsExists(query))
-		{
-			throw new AdempiereException(MSG_SETTINGS_WITH_SAME_YEAR_ALREADY_EXISTS);
-		}
-
 		final I_C_Flatrate_Conditions newConditions = flatrateBL.cloneConditionsToNewYear(conditions, newYear);
```

Comment:
```suggestion
		final I_C_Flatrate_Conditions newConditions = flatrateBL.cloneConditionsToNewYear(ConditionsId.ofRepoId(getRecord_ID()), YearId.ofRepoId(p_C_Year_ID));
```

since we're no longer doing checks at this level, please clean up the signature of the method. It can even return a TableRecordReference, or ConditionsId.